packages/panzer/adapters-stk/src/stk_interface/Panzer_STK_QuadraticToLinearMeshFactory.hpp

```diff
@@ -60,7 +61,7 @@ class QuadraticToLinearMeshFactory : public STK_MeshFactory {
 public:
 
   QuadraticToLinearMeshFactory(const std::string& quadMeshFileName,
-                          stk::ParallelMachine mpi_comm = MPI_COMM_WORLD,
```

Comment:
What's the problem with `MPI_COMM_WORLD` as a default arg?

Reply:
I wanted to align the global and default communicator for the entire package. I thought that this would be a behavior expected by the users. 
When we use `initialize_global_comm` to set a communicator for the package then this communicator will be also used as default for the objects created in the package.

This change isn't necessary so I can remove it if you want.

Reply:
@cgcgcg Is your preference to have this change reverted? 

FYI @thearusable

Reply:
I agree with @cgcgcg . The point of the effort at removing raw MPI_COMM_WORLD use is to make sure a library can run on a subset of MPI processes when given a communicator. Panzer meets this requirement as the function directly accepts a communicator. MPI_COMM_WORLD is only used as a default. All this PR is really doing is moving the use of comm world into a separate function. Nothing has changed functionally and you have only added more code to the package and more indirection (having to look up another function). The only reason it might be worth doing is if the automated script in PR testing that tries to block MPI_COMM_WORLD use is having trouble differentiating valid use cases from real issues (I would argue the tool should be fixed in that case or allow users to review and mark code as valid use).

Reply:
@thearusable Does @rppawlo 's argument apply to changes proposed or applied to other packages?

Reply:
Thanks @rppawlo . Re: The github action will ignore uses of MPI_COMM_WORLD when the comment `CHECK: ALLOW MPI_COMM_WORLD` is present on the line.

Reply:
@cgcgcg @rppawlo - I updated the PR, now the usage of `MPI_COMM_WORLD`as default parameter is allowed. 

@jwillenbring - No, only for one package for which the PR is not yet created. I will apply similar fix before creating it. 

@stmcgovern - FYI

Reply:
thanks @thearusable!